### PR TITLE
sql-statements: readability of commit/rollback

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

The rollback docs mentioned the completiontype, but didn't include the definition of it.

The mysql compatibility sections of commit and rollback talked about completiontype, but should have explained the syntax better, and the recommended alternative - since in both cases its pretty straight forward.

I have also mentioned savepoints, since it wasn't covered.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/3664
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by ran-huang** [comment]
@TomShawn Do you have a ref link?

**Comment by TomShawn** [comment]
> @TomShawn Do you have a ref link?

https://github.com/pingcap/docs/pull/3664

File: sql-statements/sql-statement-rollback.md [hunk 1]
```diff
@@ -70,6 +74,10 @@ Empty set (0.01 sec)
 
 `ROLLBACK` 语句与 MySQL 不完全兼容，TiDB 对任何 `CompletionTypeWithinTransaction` 仅有语法上的支持，即不支持事务回滚后，关闭连接或继续开启一个新事务的回滚选项。如发现任何其他兼容性差异，请在 GitHub 上提交 [issue](https://github.com/pingcap/tidb/issues/new/choose)。
```
Comment (ran-huang) [inline sql-statements/sql-statement-rollback.md:75]: The original PR deletes this paragraph. Please confirm whether to retain it in Chinese. @TomShawn

File: sql-statements/sql-statement-rollback.md [hunk 2]
```diff
@@ -70,6 +74,10 @@ Empty set (0.01 sec)
...
 
+* TiDB 不支持 savepoint 或 `ROLLBACK TO SAVEPOINT` 语法。
+* TiDB 解析但忽略 `ROLLBACK AND [NO] RELEASE` 语法。在 MySQL 中，使用该语法可在提交事务后立即断开客户端会话。在 TiDB 中，建议使用客户端程序的 `mysql_close()` 来实现该功能。
```
Comment (ran-huang) [inline sql-statements/sql-statement-rollback.md:78]:
```suggestion
* TiDB 解析但忽略 `ROLLBACK AND [NO] RELEASE` 语法。在 MySQL 中，使用该语法可在回滚事务后立即断开客户端会话。在 TiDB 中，建议使用客户端程序的 `mysql_close()` 来实现该功能。
```

File: sql-statements/sql-statement-rollback.md [hunk 3]
```diff
@@ -70,6 +74,10 @@ Empty set (0.01 sec)
...
+* TiDB 解析但忽略 `ROLLBACK AND [NO] CHAIN` 语法。在 MySQL 中，使用该语法可在提交当前事务时立即以相同的隔离级别开启新事务。在 TiDB 中，推荐直接开启新事务。
```
Comment (ran-huang) [inline sql-statements/sql-statement-rollback.md:79]:
```suggestion
* TiDB 解析但忽略 `ROLLBACK AND [NO] CHAIN` 语法。在 MySQL 中，使用该语法可在回滚当前事务时立即以相同的隔离级别开启新事务。在 TiDB 中，推荐直接开启新事务。
```

File: sql-statements/sql-statement-start-transaction.md [hunk 4]
```diff
@@ -60,12 +60,8 @@ Query OK, 0 rows affected (0.01 sec)
 
 ## MySQL 兼容性
 
-`START TRANSACTION` 语句与 MySQL 不完全兼容。
-
-* `START TRANSACTION` 相当于 MySQL 的 `START TRANSACTION WITH CONSISTENT SNAPSHOT`，即 `START TRANSACTION` 后执行了一个从 InnoDB 任意表读数据的 `SELECT` 语句（非 `SELECT FOR UPDATE`）。
-* `READ ONLY` 及其扩展选项都都只是语法兼容，其效果等同于 `START TRANSACTION`。
-
-如发现任何其他兼容性差异，请在 GitHub 上提交 [issue](https://github.com/pingcap/tidb/issues/new/choose)。
+* 执行 `START TRANSACTION` 立即在 TiDB 中开启事务。而在 MySQL 中，执行 `START TRANSACTION` 会延迟创建事务。TiDB 中的 `START TRANSACTION` 等同于 MySQL 中的 `START TRANSACTION WITH CONSISTENT SNAPSHOT`。
```
Comment (ran-huang) [inline sql-statements/sql-statement-start-transaction.md:63]: @cfzjywxk PTAL if `lazily creates a transaction` equals to `延迟创建事务`.

File: sql-statements/sql-statement-start-transaction.md [hunk 5]
```diff
@@ -60,12 +60,8 @@ Query OK, 0 rows affected (0.01 sec)
...
+* 为与 MySQL 兼容，TiDB 会解析 `START TRANSACTION READ ONLY` 语句但仍允许写入操作。
```
Comment (ran-huang) [inline sql-statements/sql-statement-start-transaction.md:64]:
```suggestion
* 为与 MySQL 兼容，TiDB 会解析 `START TRANSACTION READ ONLY` 语句，但解析后 TiDB 仍允许写入操作。
```

File: sql-statements/sql-statement-rollback.md (hunk 1 above)
Comment (TomShawn) [inline sql-statements/sql-statement-rollback.md:75]:
```suggestion
```

File: sql-statements/sql-statement-rollback.md [hunk 6]
```diff
@@ -68,7 +72,10 @@ Empty set (0.01 sec)
 
 ## MySQL 兼容性
 
-`ROLLBACK` 语句与 MySQL 不完全兼容，TiDB 对任何 `CompletionTypeWithinTransaction` 仅有语法上的支持，即不支持事务回滚后，关闭连接或继续开启一个新事务的回滚选项。如发现任何其他兼容性差异，请在 GitHub 上提交 [issue](https://github.com/pingcap/tidb/issues/new/choose)。
+
```
Comment (TomShawn) [inline sql-statements/sql-statement-rollback.md:75]:
```suggestion
```

File: sql-statements/sql-statement-start-transaction.md (hunk 4 above)
Comment (cfzjywxk) [reply sql-statements/sql-statement-start-transaction.md:63]: 执行 `START TRANSACTION` 立即在 TiDB 中开启事务。而在 MySQL 中，执行 `START TRANSACTION` 会延迟创建事务。

执行 `START TRANSACTION` 在 TiDB 中开启事务并立即生成 SNAPSHOT。而在 MySQL 中，执行 `START TRANSACTION` 会开启事务但不会立即创建 SNAPSHOT。

Comment (TomShawn) [reply sql-statements/sql-statement-start-transaction.md:63]: @cfzjywxk PTAL
```suggestion
* 执行 `START TRANSACTION` 在 TiDB 中开启事务并立即生成快照。而在 MySQL 中，执行 `START TRANSACTION` 会开启事务但不会立即生成快照。TiDB 中的 `START TRANSACTION` 等同于 MySQL 中的 `START TRANSACTION WITH CONSISTENT SNAPSHOT`。
```

File: sql-statements/sql-statement-commit.md [hunk 7]
```diff
@@ -66,7 +66,8 @@ Query OK, 0 rows affected (0.01 sec)
 
 * 在 MySQL 中，除了有多个 primary 的群组复制以外，`COMMIT` 语句通常不会导致错误。相比之下，TiDB 使用乐观并发控制，冲突可能导致 `COMMIT` 返回错误。
 * 默认情况下，`UNIQUE` 和 `PRIMARY KEY` 约束检查将延迟直至语句提交。可通过设置 `tidb_constraint_check_in_place=TRUE` 来改变该行为。
-* TiDB 对于 `CompletionTypeWithinTransaction` 仅有语法上的支持。即不支持事务提交后，关闭连接或继续开启一个新事务的提交选项。
+* TiDB 解析但忽略 `ROLLBACK AND [NO] RELEASE` 语法。在 MySQL 中，使用该语法可在提交事务后立即断开客户端会话。在 TiDB 中，建议使用客户端程序的 `mysql_close()` 来实现该功能。
```
Comment (ireneontheway) [inline sql-statements/sql-statement-commit.md:69]:
```suggestion
* TiDB 解析但忽略 `ROLLBACK AND [NO] RELEASE` 语法。在 MySQL 中，使用该语法可在提交事务后立即断开客户端会话。在 TiDB 中，建议使用客户端程序的 `mysql_close()` 来实现该。
```

Comment (ireneontheway) [inline sql-statements/sql-statement-commit.md:69]:
```suggestion
* TiDB 解析但忽略 `ROLLBACK AND [NO] RELEASE` 语法。在 MySQL 中，使用该语法可在提交事务后立即断开客户端会话。在 TiDB 中，建议使用客户端程序的 `mysql_close()` 来实现该。
```

File: sql-statements/sql-statement-commit.md [hunk 8]
```diff
@@ -66,7 +66,8 @@ Query OK, 0 rows affected (0.01 sec)
 
 * 在 MySQL 中，除了有多个 primary 的群组复制以外，`COMMIT` 语句通常不会导致错误。相比之下，TiDB 使用乐观并发控制，冲突可能导致 `COMMIT` 返回错误。
 * 默认情况下，`UNIQUE` 和 `PRIMARY KEY` 约束检查将延迟直至语句提交。可通过设置 `tidb_constraint_check_in_place=TRUE` 来改变该行为。
-* TiDB 对于 `CompletionTypeWithinTransaction` 仅有语法上的支持。即不支持事务提交后，关闭连接或继续开启一个新事务的提交选项。
+* TiDB 解析但忽略 `ROLLBACK AND [NO] RELEASE` 语法。在 MySQL 中，使用该语法可在提交事务后立即断开客户端会话。在 TiDB 中，建议使用客户端程序的 `mysql_close()` 来实现该。
```
Comment (ireneontheway) [inline sql-statements/sql-statement-commit.md:69]:
```suggestion
* TiDB 解析但忽略 `ROLLBACK AND [NO] RELEASE` 语法。在 MySQL 中，使用该语法可在提交事务后立即断开客户端会话。在 TiDB 中，建议使用客户端程序的 `mysql_close()` 来实现该功能。
```

File: sql-statements/sql-statement-commit.md (hunk 7 above)
Comment (ireneontheway) [reply sql-statements/sql-statement-commit.md:69]: Just to test whether I have the privilege to change others' PR.

File: sql-statements/sql-statement-start-transaction.md (hunk 4 above)
Comment (cfzjywxk) [reply sql-statements/sql-statement-start-transaction.md:63]: LGTM

**Comment by TomShawn** [comment]
@ran-huang @cfzjywxk All comments addressed. PTAL again, thank~

**Comment by cfzjywxk** [comment]
LGTM

**Review by ran-huang — APPROVED** [review]

**Pull request merged**